grasa_event_locator/views.py

```diff
@@ -263,52 +263,55 @@ def index(request):
         lng = []
         html_string = []
         i = 1
-
-        allEventList = Program.objects.filter(isPending=False)
-        for event in allEventList:
-                e_id.append(event.id)
-                title.append(event.title)
-                address.append(event.address)
-                coord.append(event.lat + ', ' + event.lng)
-                html_string= []
-                html_string_to_write = ""
-                i = 0
-
-        zip_table = sorted(zip(coord, e_id, title, address))
-        zip_table.sort()
-        coord, e_id, title, address = zip(*zip_table)
-        # Separate coordinates into lat and lng lists
-        # for value in coord:
-        #        lat2, lng2 = value.split(', ')
-        #        lat.append(lat2)
-        #        lng.append(lng2)
+        zip_table = ""
         try:
-                for value in coord:
-                        if i == 0:
-                                html_string_to_write = html_string_to_write + "<b>" + str(title[i]) + "</b><br>" + str(address[i]) + "<br><a href='/event/" + str(e_id[i]) + "'>Details</a><hr>"
-                        else:
-                                if str(coord[i]) == str(coord[i - 1]):
+                allEventList = Program.objects.filter(isPending=False)
+                for event in allEventList:
+                        e_id.append(event.id)
+                        title.append(event.title)
+                        address.append(event.address)
+                        coord.append(event.lat + ', ' + event.lng)
+                        html_string= []
+                        html_string_to_write = ""
+                        i = 0
+
+                zip_table = sorted(zip(coord, e_id, title, address))
+                zip_table.sort()
+                coord, e_id, title, address = zip(*zip_table)
+                # Separate coordinates into lat and lng lists
+                # for value in coord:
+                #        lat2, lng2 = value.split(', ')
+                #        lat.append(lat2)
+                #        lng.append(lng2)
+                try:
```

Comment:
try's should not be nested if avoidable.  Ideally you would only have the code that can cause the exception in the try block, and everything else outside.

Reply:
Got it. Removed the inner try block, and added the exception it was catching to the remaining try block.